genesis_core/security/verifiers/admin_bypass.py

```diff
@@ -0,0 +1,87 @@
+#    Copyright 2025 Genesis Corporation.
+#
+#    All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+# 
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import logging
+from typing import Any
+
+from restalchemy.common import contexts
+from restalchemy.common import exceptions as ra_e
+from restalchemy.dm import filters as ra_filters
+from restalchemy.dm import types as ra_types
+
+from genesis_core.user_api.iam import constants as iam_c
+from genesis_core.user_api.iam.dm import models
+from genesis_core.user_api.iam import exceptions as iam_exceptions
+from gcl_iam import exceptions as iam_e
+from gcl_iam import tokens
+from genesis_core.security.interfaces import AbstractVerifier
+
+
+log = logging.getLogger(__name__)
+
+
+class AdminBypassVerifier(AbstractVerifier):
+    """Allows admin/bypass users to skip validation.
+
+    Rule format (IamClient.rules):
+      {"kind": "admin_bypass", "bypass_users": ["admin@example.com", "<uuid>", ...]}
+    """
+
+    def __init__(self, config: dict[str, Any] = None):
+        self.config = config or {}
+
+    def can_handle(self, request) -> bool:
+        return request.headers.get("Authorization", "").startswith("Bearer ")
+
+    def verify(self, request) -> None:
+        ctx = contexts.get_context()
+        token_algorithm = ctx.context_storage.get(
+            iam_c.STORAGE_KEY_IAM_TOKEN_ENCRYPTION_ALGORITHM
+        )
+        if not token_algorithm:
+            raise iam_exceptions.CanNotCreateUser(message="Token algorithm is not configured")
+
+        auth_header = request.headers.get("Authorization", "")
+
+        try:
+            auth_token = tokens.AuthToken(
+                token=auth_header[7:].strip(),
+                algorithm=token_algorithm,
+                ignore_audience=True,
+            )
+            token = models.Token.objects.get_one(
+                filters={"uuid": ra_filters.EQ(auth_token.uuid)},
+            )
+            token.validate_expiration()
+            user = token.user
+        except (iam_e.InvalidAuthTokenError, iam_e.CredentialsAreInvalidError) as e:
+            log.debug("Admin bypass verification failed: token expired or invalid: %s", e)
+            raise iam_exceptions.CanNotCreateUser(message="Token expired or invalid")
```

Comment:
Why do we raise a specific `CanNotCreateUser` exception here? Is this AdminBypassVerifier being used for user creation only? If so, it should be renamed accordingly. If not - raised error should be a generic one.

Also, it's not clear why raising CanNotCreateUser (or similar) is better than just letting it raise iam_e.InvalidAuthTokenError / iam_e.CredentialsAreInvalidError? Debug logs are usually turned off in production. It looks like it only hides the real error, instead of adding more info to it.